### PR TITLE
Java SQLIte connector: implement getNearestMatchingAsync and getNearestMatchesAsync

### PR DESCRIPTION
**Pull request by dsgrieve**
### Motivation and Context

These methods were stubbed to return null.

### Description

Provide implementation of these methods. 

Since the methods make use of vector operations which are encapsulated in SemanticVector, the vectoroperations package and SemanticVector class were moved to the api module under com.microsoft.semantickernel.ai.embeddings

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SqliteMemoryStore.java [hunk 1]
```diff
@@ -165,17 +164,48 @@ public Mono<Void> removeBatchAsync(
...
+                                MemoryRecord record =
+                                        MemoryRecord.fromJsonMetadata(
+                                                entry.getMetadata(),
+                                                recordEmbedding,
+                                                entry.getKey(),
+                                                entry.getTimestamp());
+                                results.add(Tuples.of(record, relevanceScore));
+                            }
+                        } catch (JsonProcessingException e) {
+                            throw new RuntimeException(e);
```
Comment (johnoliver) [inline java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SqliteMemoryStore.java:200]: Is there a more specific exception for this?

File: java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/VolatileMemoryStoreTests.java [hunk 2]
```diff
@@ -93,17 +93,13 @@ void itCanCreateAndGetCollectionAsync() {
...
-        this._collectionNum++;
-
-        // Act
-        this._db.createCollectionAsync(collection).block();
+        String collection = null;
 
         // Assert
         assertThrows(
-                MemoryException.class,
+                NullPointerException.class,
```
Comment (johnoliver) [inline java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/VolatileMemoryStoreTests.java:102]: seems a bit odd to expect an NPE

Comment (dsgrieve) [reply java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/VolatileMemoryStoreTests.java:102]: This change was made to match with recent changes in .NET

File: java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SqliteMemoryStore.java (hunk 1 above)
Comment (dsgrieve) [reply java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SqliteMemoryStore.java]: changed to throw SQLConnectorException with text along the lines of "cannot deserialize database entry".

File: java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SQLiteMemoryStore.java [hunk 3]
```diff
@@ -46,62 +50,101 @@ public Mono<List<String>> getCollectionsAsync() {
...
+                                                ErrorCodes
+                                                        .ATTEMPTED_TO_ACCESS_NONEXISTENT_COLLECTION,
+                                                collectionName));
+                                return;
+                            }
+                            sink.next(exists);
+                        })
+                .then(
+                        Flux.fromIterable(records)
+                                .flatMap(record -> internalUpsertAsync(collectionName, record))
```
Comment (johnoliver) [inline java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SQLiteMemoryStore.java:132]: use concatMap to preserve order

**Review by milderhc — APPROVED** [review]

**Review by johnoliver — APPROVED** [review]

**Pull request merged**